gson/pom.xml

```diff
@@ -387,5 +387,92 @@
         </plugins>
       </build>
     </profile>
+
+    <!-- Profile for compiling a subset of Gson; can be executed with `mvn clean compile -pl gson -P gson-subset` -->
+    <profile>
+      <id>gson-subset</id>
+      <properties>
+        <gsonSubsetSrcDir>${project.build.directory}/gson-subset-src</gsonSubsetSrcDir>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-resources-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>prepare-gson-subset-src</id>
+                <phase>process-sources</phase>
+                <goals>
+                  <goal>copy-resources</goal>
+                </goals>
+                <configuration>
+                  <outputDirectory>${gsonSubsetSrcDir}</outputDirectory>
+                  <overwrite>true</overwrite>
+                  <resources>
+                    <resource>
+                      <directory>${project.build.sourceDirectory}</directory>
+                      <includes>
+                        <include>com/google/gson/stream/*</include>
+                        <include>com/google/gson/FormattingStyle.java</include>
+                        <include>com/google/gson/Strictness.java</include>
+                        <!-- Exception types -->
+                        <include>com/google/gson/JsonParseException.java</include>
+                        <include>com/google/gson/JsonIOException.java</include>
+                        <include>com/google/gson/JsonSyntaxException.java</include>
+                        <!-- Type adapter, serializer and deserializer -->
+                        <include>com/google/gson/TypeAdapter.java</include>
+                        <include>com/google/gson/JsonSerializer.java</include>
```

Comment:
I did not find I needed to include `JsonSerializer` and `JsonSerializationContext` in the subset we're using internally at Google. Are they needed here for some reason?

Reply:
No, I just guessed the subset. I have also removed `JsonDeserializer` and `JsonDeserializationContext` now. I hope the list now matches what you are using.